courses-app/src/app/features/courses/courses.module.ts

```diff
@@ -0,0 +1,13 @@
+import {NgModule} from "@angular/core";
+import {SharedModule} from "../../shared/shared.module";
+import {CoursesComponent} from "./courses.component";
+import {FontAwesomeModule} from "@fortawesome/angular-fontawesome";
+import {ButtonComponent} from "../../shared/components/button/button.component";
+
+
+@NgModule({
+  imports: [SharedModule, FontAwesomeModule,],
+  declarations: [CoursesComponent, ButtonComponent],
```

Comment:
ButtonComponent belongs to the shared.module so should be declared and exported there. You are importing it with SharedModule in the line before.